docker/demo-indirect-invocation.yml

```diff
@@ -1,17 +1,21 @@
 services:
   demo-indirect-invocation:
-    image: integritee-cli:dev
-    container_name: integritee-indirect-invocation-demo
+    image: integritee-cli:${VERSION:-dev}
+    devices:
+      - "/dev/sgx/provision:/dev/sgx/provision"
+      - "/dev/sgx/enclave:/dev/sgx/enclave"
+    volumes:
+      - "/var/run/aesmd:/var/run/aesmd"
```

Comment:
Would this also work on non-sgx-machines if the worker is compiled in software mode?

Reply:
I do not think so, because of this:

```bash
[~]  $ docker run --device /dev/non_existing_device hello-world
docker: Error response from daemon: error gathering device information while adding custom device "/dev/non_existing_device": no such file or directory.
ERRO[0000] error waiting for container:    
```

Reply:
Hmm, then I have some issues with this approach. In my opinion, the docker demos must be buildable/runnable on non-sgx machines.

Reply:
The solution is:
    devices:
      - "${SGX_PROVISION:-/dev/null}:/dev/sgx/provision"
      - "${SGX_ENCLAVE:-/dev/null}:/dev/sgx/enclave"
    volumes:
      - "${AESMD:-/dev/null}:/var/run/aesmd"